analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Operations.cs

```diff
@@ -367,6 +367,34 @@ public void Conversion_ToLocalNonNullableValueType_VB()
         validator.TagValue("TryCastBoxing").Should().HaveOnlyConstraints(ObjectConstraint.NotNull, DummyConstraint.Dummy);
     }
 
+    [TestMethod]
+    public void Conversion_TryDownCast_Branches()
+    {
+        var validator = SETestContext.CreateCS("""
+                var conversion = (arg as Exception);
+                Tag("Arg", arg);
+                Tag("Conversion", conversion);
+                """, "object arg").Validator;
+        validator.TagValues("Arg").Should().SatisfyRespectively(
+            x => x.Should().BeNull(),
+            x => x.Should().BeNull());
+        validator.TagValues("Conversion").Should().SatisfyRespectively(
+            x => x.Should().HaveOnlyConstraint(ObjectConstraint.Null),
+            x => x.Should().HaveOnlyConstraint(ObjectConstraint.NotNull));
+    }
+
+    [TestMethod]
+    public void Conversion_TryUpCast_DoesNotBranch()
+    {
+        var validator = SETestContext.CreateCS("""
+                var conversion = (arg as Exception);
+                Tag("Arg", arg);
+                Tag("Conversion", conversion);
+                """, "ArgumentException arg").Validator;
+        validator.TagValues("Arg").Should().SatisfyRespectively(x => x.Should().BeNull());
+        validator.TagValues("Conversion").Should().SatisfyRespectively(x => x.Should().BeNull());
```

Comment:
Using `TagValue` asserts that you expect a single one
```suggestion
        validator.TagValue("Arg").Should().BeNull();
        validator.TagValue("Conversion").Should().BeNull();
```